### DISCUSSION TITLE
Replacing apache and .htaccess for routing

### DISCUSSION THREAD

**Discussion by JWasdin** [brefphp/bref#939] [Support] [answered]
I am attempting to deploy [Sendy](https://sendy.co) (a self-hosted PHP mail app) using Bref, and Sendy uses apache as the webserver and an `.htaccess` file to handle some URL routing. While I was able to get the base code to deploy via Bref it resulted in a known [redirect loop error](https://sendy.co/troubleshooting#getting-redirected-over-and-over-to-login-page-infinite-loop). The recommended fix is tweaking the `.htaccess` file and ensure the app is deployed on apache. 

I know that Bref is using API Gateway + Lambda as the alternative for handling and serving requests based on the docs [here](https://bref.sh/docs/runtimes/http.html#routing). As there is no apache webserver when using Bref I'm looking for examples or ideas of how to handle a PHP app that is expecting apache when using Bref. Any help or pointers would be greatly appreciated.

Many thanks!

**Comment by mnapoli** [comment, marked as answer]
Hi! To be honest, it might be hard to find a solution without digging into _why Sendy redirects_.

I would expect Sendy to expect some headers (the URL, HTTPS, etc.) to be in a certain way. If it isn't, Sendy issues a redirect to another URL.

What you could do to trick Sendy is to edit `index.php` (or whatever that filename is) to manually override `$_SERVER` (or any other necessary variable) with the values that Sendy expects.

I don't know how to help more than that unfortunately.

Oh, one other possibility is that Sendy doesn't work with a single `index.php` entry file, but with multiple public files. In that case that might be even harder to run on Lambda.

**Reply by JWasdin** [reply]
Alright, thanks for the response. It seems that Sendy just may not be compatible with running serverless at this point in time.
